newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js

```diff
@@ -0,0 +1,147 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { type ParameterInlineRendererProps } from './ParameterInlineRenderer.flow';
+import React from 'react';
+import { type ParameterFieldProps } from './ParameterFieldCommons';
+import SelectField from '../../UI/SelectField';
+import SelectOption from '../../UI/SelectOption';
+import { TextFieldWithButtonLayout } from '../../UI/Layout';
+import RaisedButton from '../../UI/RaisedButton';
+import { type Leaderboard } from '../../Utils/GDevelopServices/Play';
+import LeaderboardContext from '../../Leaderboard/LeaderboardContext';
+import OpenInNew from '@material-ui/icons/OpenInNew';
+import { t } from '@lingui/macro';
+import Toggle from '../../UI/Toggle';
+import { LargeSpacer } from '../../UI/Grid';
+import LeaderboardDialog from '../../SceneEditor/LeaderboardDialog';
+import GenericExpressionField from './GenericExpressionField';
+
+const getLeaderboardNameByValue = (
+  leaderboards: ?Array<Leaderboard>,
+  value: string
+): string => {
+  if (!leaderboards) return value;
+  const quotedIdLeaderboards = leaderboards.map(leaderboard => ({
+    id: `"${leaderboard.id}"`,
+    name: leaderboard.name,
+  }));
+  const leaderboard = quotedIdLeaderboards.find(
+    leaderboard => leaderboard.id === value
+  );
+  return leaderboard ? leaderboard.name : value;
+};
+
+export function LeaderboardIdField(props: ParameterFieldProps) {
+  const { leaderboards } = React.useContext(LeaderboardContext);
+  const [isAdminOpen, setIsAdminOpen] = React.useState(false);
```

Comment:
Could we have a React effect that reloads the leaderboard if we have none loaded? Or even if they are loaded, it's ok to update them silently: